### PR TITLE
Bump min sdk version to 26

### PR DESCRIPTION
**Pull request by wzieba**
## Description

This PR bumps min sdk version to 26 and addresses lint errors, that raised after the bump. Internal reference: p1720698865814969-slack-C04PWEZSYFL

## To Test:

This PR shouldn't change behavior changes. To test, make sure that on both WordPress and Jetpack apps:

- icons of the app look good
  - for Jetpack, verify "Themed icons" feature in both dark and light modes. Details: #19643
- shortcuts (long press on the app icon) icons look as expected too
- smoke test some vital features

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverServiceStarter.kt [hunk 1]
```diff
@@ -22,30 +19,22 @@ object ReaderDiscoverServiceStarter {
...
+            .setOverrideDeadline(0) // if possible, try to run right away
+            .setExtras(extras)
+            .build()
+        val jobScheduler = context.getSystemService(Context.JOB_SCHEDULER_SERVICE) as JobScheduler
+        val resultCode = jobScheduler.schedule(jobInfo)
+        if (resultCode == JobScheduler.RESULT_SUCCESS) {
+            AppLog.i(READER, "reader discover service > job scheduled")
         } else {
-            // schedule the JobService here for API >= 26. The JobScheduler is available since API 21, but
-            // it's preferable to use it only since enforcement in API 26 to not break any old behavior
```
Comment (ParaskP7) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverServiceStarter.kt:31]: Minor (🔍): On all other such changes you made, chose to keep a similar comment. Maybe we should keep this comment as well? 😊

File: WordPress/src/jetpack/res/xml/shortcuts.xml [hunk 2]
```diff
@@ -1,7 +1,5 @@
 <!--WARNING - build flavors might use a copy of this file -->
-<shortcuts xmlns:android="http://schemas.android.com/apk/res/android"
-           xmlns:tools="http://schemas.android.com/tools"
-           tools:targetApi="n_mr1">
```
Comment (ParaskP7) [inline WordPress/src/jetpack/res/xml/shortcuts.xml:4]: Warning (⚠️): This change you did here also applies to other variants, please search for `tools:targetApi="n_mr1"` and apply the same change there too:

- `wasabi/.../shortcuts.xml`
- `jetpackWasabi/.../shortcuts.xml`
- `jalapeno/.../shortcuts.xml`
- `jetpackJalapeno/.../shortcuts.xml`

**Review by ParaskP7 — CHANGES_REQUESTED** [review]
👋 @wzieba !

I have reviewed and tested this PR as per the instructions, everything works as expected, good job! 🌟 

-----

I have left a warning (⚠️) for you and a minor (🔍) comment. I am **NOT** going to approve this PR, for now, but there is nothing serious about it.

-----

`EXTRAS`
1. Warning (⚠️): We have been also discussing about retained APKs. Consider taking this opportunity to verify if we want to actually retain the `minSdk = 24` APK and if so, do apply the relevant `Fastfile` change (context -> p1720818389401089/1720698865.814969-slack-C04PWEZSYFL). If not, just ignore this warning.

**Comment by ParaskP7** [comment]
FYI: I also asked this pcdRpT-499-p2#comment-10436 internal question about themed icons, I was quite surprised that this was not working for the `WordPress` app too. 🤷 🤔

File: WordPress/src/jetpack/res/xml/shortcuts.xml (hunk 2 above)
Comment (wzieba) [reply WordPress/src/jetpack/res/xml/shortcuts.xml:4]: Thanks! Addressed in e42dd6d

File: WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverServiceStarter.kt (hunk 1 above)
Comment (wzieba) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverServiceStarter.kt:31]: Good point - actually, I decided to go with removing this comment in other places as this is explanation why we don't use JobScheduler in API 21+, even though it was already available. And this code is now removed. Addressed in f375b75

**Comment by wzieba** [comment]
Thanks for the review @ParaskP7 !

> FYI: I also asked this pcdRpT-499-p2#comment-10436 internal question about themed icons, I was quite surprised that this was not working for the WordPress app too. 🤷 🤔

Good point, thanks!

File: WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverServiceStarter.kt (hunk 1 above)
Comment (ParaskP7) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverServiceStarter.kt:31]: Great @wzieba , yea, as long as it is consistent I am okay, but I would rather have that comment removed as well, so thanks! 👍

File: WordPress/src/jetpack/res/xml/shortcuts.xml (hunk 2 above)
Comment (ParaskP7) [reply WordPress/src/jetpack/res/xml/shortcuts.xml:4]: Awesome, thanks @wzieba ! 👍

**Comment by ParaskP7** [comment]
> Warning (⚠️): We have been also discussing about retained APKs. Consider taking this opportunity to verify if we want to actually retain the minSdk = 24 APK and if so, do apply the relevant Fastfile change (context -> p1720818389401089/1720698865.814969-slack-C04PWEZSYFL). If not, just ignore this warning.

Btw, what do you think we should do about this @wzieba ? 🤔

**Comment by wzieba** [comment]
I think we should retain, but let's do this during the next release cycle (by release manager) as I'm not sure when this PR will be released, so I can't tell the version code to retain.

**Comment by ParaskP7** [comment]
> I think we should retain...

👍 @wzieba 

> ...but let's do this during the next release cycle (by release manager)

I fear that us doing that during the next release cycle will be too late. 🤔

To my understanding the next release is going to be the `25.2` for JP/WPAndroid. There hasn't been any code-freezed branch yet (ie. `release/25.2`).

> ...as I'm not sure when this PR will be released, so I can't tell the version code to retain.

Per the above, when this PR gets merged, if the retained configuration is added, it should apply to the next release.

So, based on what I am currently seeing in production (`Google Play Console`), the (potential) version to retain is the `1440` for `25.1` (see also: [version.properties](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/version.properties)).

Cc @oguzkocer to help us here with release management. 🙏

**Comment by oguzkocer** [comment]
> Cc @oguzkocer to help us here with release management. 🙏

@ParaskP7 What exactly can I help you with?

**Comment by ParaskP7** [comment]
👋 @oguzkocer ! 😊

> @ParaskP7 What exactly can I help you with?

To confirm that my response to @wzieba is correct, and that if we want to retain the latest APK release (`1440` -> `minSdk = 24`), that we indeed need to add the `version_codes_to_retain: [1440]` Fastlane configuration here and now. 🙏 

Else, we might forget doing that, risking releasing with `minSdk = 26` but without retaining the previous APK release (`1440` -> `minSdk = 24`). Wdyt?

**Comment by oguzkocer** [comment]
It's best to merge this PR with a version code to retain, but unfortunately it's not possible to give a definitive answer of what that version is. I'd say put in your best guess, which is probably the current `25.1` release, and I'll try and remember to update it if we release anything else until these changes are released.

In all likelihood, we'll do the next release from `trunk`, in which case there won't be any problems. However, if we do a hotfix release or a small release that's branched from `25.1`, we will need to update the value.

Hope that answers your question!

**Comment by ParaskP7** [comment]
Thanks for your input @oguzkocer !

> In all likelihood, we'll do the next release from trunk, in which case there won't be any problems.

👍 

> However, if we do a hotfix release or a small release that's branched from 25.1, we will need to update the value.

If we branch from `25.1`, then it will still be `minSdk = 24`, right? Thus, we wouldn't have any problems anyway, no need to worry about `minSdk = 26` or any configuration on retaining an APK, isn't that correct? 🤔 

Apologies, I might be misunderstanding something... ☺️

**Review by ParaskP7 — APPROVED** [review]
🎉 😊 🚀

**Comment by oguzkocer** [comment]
> If we branch from `25.1`, then it will still be `minSdk = 24`, right? Thus, we wouldn't have any problems anyway, no need to worry about `minSdk = 26` or any configuration on retaining an APK, isn't that correct? 🤔 

It won't be a problem for that release, however the latest `minSdk = 24` release would have moved. Assuming that we want to retain the latest possible `minSdk = 24` version when we update it to `26`, we would want to update the retained version code if we release a hotfix version.

Does that make sense? I am happy to provide a detailed example if it's not clear.

**Comment by ParaskP7** [comment]
Hmmm, I think I understand @oguzkocer , it all depends on whether we would need to hotfix something and the state of our APKs at that moment, I think I don't need for further details, thanks! 🙇

**Pull request merged**